src/go/k8s/pkg/resources/certmanager/kafka_api.go

```diff
@@ -81,8 +80,25 @@ func (r *PkiReconciler) prepareKafkaAPI(
 		if externalListener != nil && externalListener.External.Subdomain != "" {
 			dnsName = append(dnsName, externalListener.External.Subdomain)
 		}
-
-		redpandaCert := NewNodeCertificate(r.Client, r.scheme, r.pandaCluster, certsKey, nodeIssuerRef, dnsName, cn, false, keystoreSecret, r.logger)
+		// The use of the common name field has been deprecated since 2000 and is discouraged from being used.
+		// Nevertheless, if the commonName field is omitted, the first element in the list of SANs will be set as the common name by cert-manager.
+		// In the case of Redpanda, since we're issuing a wildcard certificate for all the nodes, the commonName must match the Subdomain.
+		emptyCommonName := CommonName("")
+		// TODO: validate that len(dnsName[0]) <= 63
+		// Some rules that apply:
+		// - It represents the server name protected by the SSL certificate. The certificate is valid only if the request hostname matches the certificate common name. Most web browsers display a warning message when connecting to an address that does not match the common name in the certificate.
+		// - In the case of a single-name certificate, the common name consists of a single host name (e.g. example.com, www.example.com), or a wildcard name in case of a wildcard certificate (e.g. *.example.com).
+		// - The common name is technically represented by the commonName field in the X.509 certificate specification.
+		// - The common name is limited to 63 characters. It is an obsolete method of indicating a domain name to which the certificate applies, since current Internet standards expect software to check only the Subject Alternative Names in order to determine the applicability of a certificate.
+		// - It must precisely match the server name where the certificate is installed. If the certificate is issued for a subdomain, it should be the full subdomain. For instance, for the www and api subdomains of example.com, the common name will be www.example.com or api.example.com, and not example.com.
+		// - The common name can only contain up to one entry: either a wildcard or non-wildcard name. It’s not possible to specify a list of names covered by an SSL certificate in the common name field.
+		// - The Subject Alternative Name extension (also called Subject Alternate Name or SAN) was introduced to solve this limitation. The SAN allows issuance of multi-name SSL certificates.
+		// - The CA/Browser Forum has since mandated that the SAN would also include any value present in the common name, effectively making the SAN the only required reference for a certificate match with the server name. The notion of the common name survives mostly as a legacy of the past. There are active discussions to remove its use from most browsers and interfaces.
```

Comment:
Can you shorten the comments. I have wide screen and it doesn't display correctly on my end. Maybe 100-120 characters maks.